diff --git a/src/regex_impl.cc b/src/regex_impl.cc
index de00ce87..d27085ac 100644
--- a/src/regex_impl.cc
+++ b/src/regex_impl.cc
@@ -878,16 +878,7 @@ void dump_regex(const CompiledRegex& program)
 
 CompiledRegex compile_regex(StringView re)
 {
-    CompiledRegex res;
-    try
-    {
-        res = RegexCompiler::compile(re);
-    }
-    catch (runtime_error& err)
-    {
-        write_to_debug_buffer(err.what());
-    }
-    return std::move(res);
+    return RegexCompiler::compile(re);
 }
 
 auto test_regex = UnitTest{[]{
